app/models/auction.rb

```diff
@@ -107,20 +108,23 @@ def in_progress?
     end
   end
 
-  def in_progress_by_date?(date)
-    (starts_at <= date.end_of_day || starts_at <= date.beginning_of_day) &&
-      (ends_at >= date.end_of_day || ends_at >= date.beginning_of_day)
+  def find_auction_turns
+    self.turns_count = calculate_turns_count
   end
 
-  def search_data
-    {
-      id: id,
-      domain_name: domain_name,
-      created_at: created_at,
-      starts_at: starts_at,
-      ends_at: ends_at,
-      offers_count: offers.count,
-      completed: result.present?,
-    }
+  def calculate_turns_count
+    auctions = Auction.unscoped.where(domain_name: domain_name).where('starts_at <= ?', starts_at)
```

Comment:
Wondering if this could be moved to scope instead so it can be used by potential other developments? Or implementing 2 params for scope function would be too "hacky"?

Reply:
Not 'hacky' at all, but I prefer not to pre-improve.
Suppose we'll move it to scope when it will be any other place it needed. Until when make a scope for a single usage seems like overkill to me.